src/pages/ImportXml/index.tsx

```diff
@@ -98,8 +51,27 @@ function ImportXml() {
         importTime: elem.importTime ? `${elem.importTime}s` : '',
         user: elem.user,
       }));
-      setPageState(currentValue => ({ ...currentValue, total: data.totalElements }));
+      const reversedXmls = [...xmls].reverse();
+      setFirstRowByProfessor(prevState => {
```

Comment:
Being the first row for a professor doesn't mean that the XML is in the backend.

Reply:
I didn't yet understand the problem here, but for those lines of code I noticed that the xmls retrieved from the backend are only by page, and also they come in order of oldest to most recent, so I reversed them only to get the most recent ones that could be reprocessed, and also this line of code "setPageState(currentValue => ({ ...currentValue, total: data.totalElements }));" remains the same, I just moved it to the end of the function.

Reply:
The problem here is that you are using this to show the icon and enable reprocessing the XML. But it dont grant that there is a XML to reprocess.
The best approach is to return information from the backend whether it is possible to reprocess or not.

Reply:
So in this case, I should add a new field/column in the importXML SQL table that says if it is reprocessable or not?

Reply:
That would do just fine.
Imagine the following cenario: This branch gets in production as it is. And there is no XML in backend. The frontend will say that it is possible to reprocess every professor, but that is not true.
The information should come from the backend. And the only ones that should be able to reprocess are the ones that have the file, i.e., the ones that are saved from the new backend version.
But since we will currently save only the most recent and updated XML, then you will need to disable the last ones when a new one is uploaded.